federation/pkg/federation-controller/deployment/deploymentcontroller_test.go

```diff
@@ -174,6 +185,23 @@ func TestDeploymentController(t *testing.T) {
 	deploymentsWatch.Add(dep2)
 	assert.NoError(t, CheckObjectFromChan(cluster1CreateChan, checkDeployment(dep2, 6)))
 	assert.NoError(t, CheckObjectFromChan(cluster2CreateChan, checkDeployment(dep2, 3)))
+
+	// Add new deployment with non-default replica placement preferences.
+	dep3 := newDeploymentWithReplicas("deployment3", 6)
+	dep3.Annotations = make(map[string]string)
+	dep3.Annotations[FedDeploymentPreferencesAnnotation] = `{"rebalance": true,
+		  "clusters": {
+		    "*": {"weight": 1}
```

Comment:
Can you add tests for min and max replicas as well?

Also do we need these tests at deployment controller level? Can we test the scheduling library independently? We can have more exhaustive test cases there and since deployment, replicasets and soon jobs will use the same shared scheduling code, we wont have to duplicate the tests in all controllers.